v2/internal/frontend/desktop/common/uri_translate.go

```diff
@@ -1,20 +1,34 @@
 package common
 
-import "net/url"
+import (
+	"fmt"
+	"net/url"
+)
 
-func TranslateUriToFile(uri string, expectedScheme string, expectedHost string) (file string, match bool, err error) {
+var ErrUnexpectedScheme = fmt.Errorf("unexpected scheme")
+var ErrUnexpectedHost = fmt.Errorf("unexpected host")
+
+func translateUriToFile(uri string, expectedScheme string, expectedHosts ...string) (file string, err error) {
 	url, err := url.Parse(uri)
 	if err != nil {
-		return "", false, err
+		return "", err
 	}
 
-	if url.Scheme != expectedScheme || url.Host != expectedHost {
-		return "", false, nil
+	if url.Scheme != expectedScheme {
+		return "", ErrUnexpectedScheme
 	}
 
-	filePath := url.Path
-	if filePath == "" {
-		filePath = "/"
+	for _, expectedHost := range expectedHosts {
```

Comment:
On some platforms, I've seen a nil host come through. Is there a reason to consider the host at all?

Reply:
On Linux and Macos the callback is only called for the wails:// scheme, so we could map all hosts to the asset fs if we want to.
On Windows the callback is called for all requests and the assets are under file://wails. Currently we let handle all other file:// requests by the default handler by just returning from the callback. So thats to detect this case, but we could also detect that case in the windows callback itself.

Reply:
I think for the case mentioned on the slack thread it would get more information for the user if he knows that the host was wrong and a 500 is returned than just a 404.
In this implementation the log entry specificly mentions the host was wrong.

Reply:
I was holding off until Linux beta was released then it slipped off the radar. Will try testing today.